src/elf_parser.cpp

```diff
@@ -0,0 +1,290 @@
+/**
+ * @file elf_parser.cpp
+ * @author Michael Chan
+ * @brief ELF file parser implementation file
+ * @version 0.1
+ * @date 2025-07-17
+ *
+ * @copyright Copyright (c) 2025
+ *
+ **/
+
+#include "../include/elf_parser.hpp"
+
+ElfParser::ElfParser(std::string_view p_file_name)
+  : m_file_name(p_file_name)
+{
+    try {
+        if (elf_version(EV_CURRENT) == EV_NONE) {
+            throw std::runtime_error(
+              "ELF library initialization failed. Wrong ELF format version.");
+        }
+    } catch (const std::runtime_error& e) {
+        std::println(stderr, "Error opening {}: {}", m_file_name, e.what());
+        exit(EXIT_FAILURE);
+    }
+
+    m_file = open(m_file_name.c_str(), O_RDONLY, 0);
```

Comment:
Not sure how cross platform this will, may need to change it to std::filesystem or something. Its fine for right now.